frontend/javascripts/admin/dataset/composition_wizard/04_configure_new_dataset.tsx

```diff
@@ -0,0 +1,285 @@
+import { OxalisState } from "oxalis/store";
+import React, { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { APIDataLayer, APIDataset, APIDatasetId, APITeam, LayerLink } from "types/api_flow_types";
+import { syncValidator } from "types/validation";
+import { WizardComponentProps } from "./common";
+
+const FormItem = Form.Item;
+
+export function ConfigureNewDataset(props: WizardComponentProps) {
+  const formRef = React.useRef<FormInstance<any>>(null);
+
+  const onPrev = () => {
+    props.setWizardContext((oldContext) => ({
+      ...oldContext,
+      currentWizardStep: "SelectDatasets",
+    }));
+  };
+
+  const [isLoading, setIsLoading] = useState(false);
+  const activeUser = useSelector((state: OxalisState) => state.activeUser);
+  const isDatasetManagerOrAdmin = Utils.isUserAdminOrDatasetManager(activeUser);
+  const [form] = Form.useForm();
+  const [selectedTeams, setSelectedTeams] = useState<APITeam | Array<APITeam>>([]);
+
+  const { wizardContext } = props;
+  const linkedDatasets = wizardContext.datasets;
+
+  const onRemoveLayer = (layer: LayerLink) => {
+    const oldLayers = form.getFieldValue(["layers"]);
+    const newLayers = oldLayers.filter((existingLayer: LayerLink) => existingLayer !== layer);
+    form.setFieldsValue({ layers: newLayers });
+  };
+
+  const handleTransformImport = async (sourcePoints: Vector3[], targetPoints: Vector3[]) => {
+    const datasets = linkedDatasets;
+    const transformationArr =
+      sourcePoints.length > 0 && targetPoints.length > 0
+        ? [
+            {
+              type: "affine" as const,
+              matrix: flatToNestedMatrix(estimateAffineMatrix4x4(sourcePoints, targetPoints)),
+            },
+          ]
+        : [];
+
+    const newLinks: LayerLink[] = (
+      _.flatMap(datasets, (dataset) =>
+        dataset.dataSource.dataLayers.map((layer) => [dataset, layer]),
+      ) as [APIDataset, APIDataLayer][]
+    ).map(
+      ([dataset, dataLayer]): LayerLink => ({
+        datasetId: {
+          owningOrganization: dataset.owningOrganization,
+          name: dataset.name,
+        },
+        sourceName: dataLayer.name,
+        newName: dataLayer.name,
+        transformations: dataset === datasets[0] ? transformationArr : [],
+      }),
+    );
+    form.setFieldsValue({ layers: newLinks });
+  };
+
+  useEffect(() => {
+    handleTransformImport(wizardContext.sourcePoints, wizardContext.targetPoints);
+  }, []);
+
+  const handleSubmit = async () => {
+    if (activeUser == null) {
+      throw new Error("Cannot upload dataset without being logged in.");
+    }
+    const layers = form.getFieldValue(["layers"]);
+
+    const uploadableDatastores = props.datastores.filter((datastore) => datastore.allowsUpload);
+    const datastoreToUse = uploadableDatastores[0];
+    if (!datastoreToUse) {
+      Toast.error("Could not find datastore that allows uploading.");
+      return;
+    }
```

Comment:
Is this correct? Can datasets only be created (using this wizard) on data stores that allow upload?

Reply:
I think so, because the datastore will create a new folder with symlinks. This is probably only possible if uploads are allowed. Maybe @fm3 can confirm this?

Reply:
Kinda. The datastore at least needs write access on its file system. This is usually the case for those datastores with allowUpload=true. There may be cases where this is too strict (some datastores could allow this, but we don’t want users to upload data there), but let’s introduce more fine-grained logic when we find that it is needed.